### DISCUSSION TITLE
I just used this library to successfully un-corrupt a corrupted project file.

### DISCUSSION THREAD

**Discussion by gwennlbh** [demberto/PyFLP#184] [Show and tell]
I can't overstate how grateful I am that this worked:

```python
import pyflp

project = pyflp.parse('corrupted.flp')
pyflp.save(project, 'uncorrupted.flp')
```

and just like that -- I recovered hours of work.

Thank you so much.

You might as well advertise that this project can sometimes un-brick FLP files ^^

I can give you the project files if you want to analyze why it worked

**Comment by demberto** [comment]
Uhh, that's not actually supposed to happen. Just saving it back should give you the same exact file bit-by-bit. I'd love to look into what might have made this happen, but currently I don't have enough time.

**Reply by amd75692** [reply]
Would it be possible to somehow recover a corrupted file ?

**Reply by demberto** [reply]
@amd75692 No.
